zconfig_test.go

```diff
@@ -46,8 +48,37 @@ type M struct {
 	}
 }
 
+type Tester struct {
+	t *testing.T
+}
+
+func (tester *Tester) Hook(field *Field) error {
+	expectedProviders := map[string]string{
+		"foo": "test",
+		"baz": "test2",
+		"quz": "default",
+	}
+
+	expectedProvider, found := expectedProviders[field.Key]
+	if !found {
+		return nil
+	}
+
+	if field.Provider == "" {
```

Comment:
I think this condition is redundant. Also, you should quote the values for the next one `unexpected provider for key %q, got %q, expected %q` so the empty case is easier to read.

Reply:
Never used %q, thank you